task_processing/plugins/kubernetes/kubernetes_pod_executor.py

```diff
@@ -392,36 +392,47 @@ def _pending_event_processing_loop(self) -> None:
 
         logger.debug("Exiting Pod event processing - stop requested.")
 
+    def _create_container_definition(
+        self,
+        name: str,
+        task_config: KubernetesTaskConfig,
+    ) -> V1Container:
+        return V1Container(
+            image=task_config.image,
+            name=name,
```

Comment:
i think we'll want to go ahead and sanitize this name (what i was trying to be lazy and avoid doing by naming the original container "main") :p 

container names need to be valid [RFC 1123](https://www.rfc-editor.org/rfc/rfc1123#section-2) DNS labels - https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/vendor/sigs.k8s.io/kustomize/kyaml/yaml/internal/k8sgen/pkg/util/validation/validation.go?L190 is what k8s does right now (tl;dr: 63 char max and is captured by this regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?`

we should be able to use [get_sanitised_kubernetes_name](https://sourcegraph.yelpcorp.com/Yelp/task_processing@8ab2a98/-/blob/task_processing/plugins/kubernetes/utils.py#L101:5) to do this tho

Reply:
that said - i'm not particularly sure what i'd prefer: that we truncate (i.e., use `get_sanitised_kubernetes_name` or that we reject too-long names at validation. 

i think i'd lean towards truncation? it's what we do pretty much everywhere else, so seems fine to continue doing that here

Reply:
I think if people aren't actually going to refer to the names exactly, then it should be fine?